cpp/src/copying/gather.cu

```diff
@@ -29,6 +29,8 @@
 #include <cuda/functional>
 #include <thrust/iterator/transform_iterator.h>
 
+#include <stdexcept>
```

Comment:
Should we be including `<stdexcept>` in a lot more files in this PR? I assume it's being pulled in by some transitive includes everywhere else.

Reply:
I like IWYU. Let's do it.